### PR TITLE
chrome: update to 89.0.4389.114 and fix gtk3 compile issues and addon (105)

### PR DESCRIPTION
**Pull request by heitbaum**
### Included changes:
Note: these are only chrome dependancies - so no impact to other LE10 core or addons
- gdk-pixbuf: update to 2.42.4
- gtk3: disable use of to-pixdata during build
- gtk3: update to 3.24.28
- pango: update to 1.48.4
- at-spi2-core: update to 2.40.0
- chrome: update to 89.0.4389.114 and fix gtk3 compile issues and addon (105)
- New dependency for chrome 89 versus 87 - chrome-libxshmfence

### Notes:
- harfbuzz bump is not included as it is also used by libass
- Build tested against master [HEAD]
- Run tested on TGL

### CODE REVIEW

**Comment by mglae** [comment]
```
Apr 04 18:05:40.951241 le10mars kodi.sh[1448]: /storage/.kodi/addons/browser.chrome/chrome-bin/chrome: error while loading shared libraries: libxshmfence.so.1: cannot open shared object file: No such file or directory
```
Without the bump chrome 87.0.4280.66 is starting.

**Comment by heitbaum** [comment]
Added chrome-libxshmfence, running full build and run test now.

File: packages/addons/addon-depends/chrome-depends/chrome-libxshmfence/package.mk [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+# Copyright (C) 2021-present Team LibreELEC (https://libreelec.tv)
+
+. $(get_pkg_directory libxshmfence)/package.mk
+
+PKG_NAME="chrome-libxshmfence"
+PKG_LONGDESC="libxshmfence for chrome"
+PKG_URL=""
+PKG_DEPENDS_UNPACK+=" libxshmfence"
+PKG_BUILD_FLAGS="-sysroot"
+PKG_TOOLCHAIN="autotools"
```
Comment (CvH) [inline packages/addons/addon-depends/chrome-depends/chrome-libxshmfence/package.mk:11]: can you remove ? works for me without
```suggestion
```

Comment (heitbaum) [reply packages/addons/addon-depends/chrome-depends/chrome-libxshmfence/package.mk]: Yup. All good.

**Review by CvH — APPROVED** [review]

**Comment by heitbaum** [comment]
Whilst the above PR fixed the build on local build hosts - it is failing on the Jenkins build server in either itstool or shared-mime-info.
Failed build output - http://paste.ubuntu.com/p/Zv6F6ZvHNr/
A successful build - http://paste.ubuntu.com/p/Gbm6bkth6D/

**Comment by heitbaum** [comment]
### Notes:

thinking the issue is around **shared-mime-info-gmo**
`find build.LibreELEC-Generic.x86_64-10.0-devel/build/shared-mime-info-2.1/.x86_64-libreelec-linux-gnu/ -type f -exec grep -H shared-mime-info-gmo {} \; `


### See the build target in:
- shared-mime-info-2.1/.x86_64-libreelec-linux-gnu/build.ninja

looking at the code below the `shared-mime-info-gmo` target needs python, and based on the following article https://unix.stackexchange.com/questions/240702/import-unable-to-open-x-server-error-import-c-importimagecommand-361 it may be that the Jenkins build server does not have the right environment that shared-mime-info is expecting python (as **from** and **import** look to be being executed by **sh**. Guessing to be a bug/issue with the meson build framework of shared-mime-info.

```
build shared-mime-info-gmo: phony meson-shared-mime-info-gmo

build meson-shared-mime-info-gmo: CUSTOM_COMMAND
 COMMAND = /storage/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/bin/meson --internal exe --unpickle /storage/home-rudi/LibreELEC.tv/build.LibreELEC-Generic.x86_64-10.0-devel/build/shared-mime-info-2.1/.x86_64-libreelec-linux-gnu/meson-private/meson_exe_meson_2d27ed4593f17fa092e43bf14fd7ef397f7a586a.dat
 description = Running$ external$ command$ shared-mime-info-gmo$ (wrapped$ by$ meson$ to$ set$ env)
 pool = console
```

**Comment by SupervisedThinking** [comment]
well does the chrome addon build python3 when you build it standalone?
```
/home/jails/home/jenkins/LE/build2/workspace/Addons/All_Addons-Generic/LibreELEC.tv/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/bin/itstool: 36: /home/jails/home/jenkins/LE/build2/workspace/Addons/All_Addons-Generic/LibreELEC.tv/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/bin/itstool: PY3: not found
/home/jails/home/jenkins/LE/build2/workspace/Addons/All_Addons-Generic/LibreELEC.tv/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/bin/itstool: 39: /home/jails/home/jenkins/LE/build2/workspace/Addons/All_Addons-Generic/LibreELEC.tv/build.LibreELEC-Generic.x86_64-10.0-devel/toolchain/bin/itstool: Syntax error: "(" unexpected (expecting "then")
```

**Comment by heitbaum** [comment]
You might be right - it is as if pythonX is not available to meson/itstool. As the itstool we are using is itstool:host. 

Is this the fix? Same as heimdal

```
-PKG_DEPENDS_TARGET="toolchain glib libxml2 gettext itstool:host"
+PKG_DEPENDS_TARGET="toolchain glib libxml2 gettext itstool:host Python3:host"
```

**Comment by heitbaum** [comment]
### Last runs
533 - success - build5
534 - failed - build2
535 - success - build5
536 - failed - build2
537 - success - build5
538 - failed - build2

**Comment by CvH** [comment]
build2 Debian SID
build5 Ubuntu 18.04

**Comment by heitbaum** [comment]
> build2 Debian SID
> build5 Ubuntu 18.04

Setting up test environments for all docker variants now to hone in on build issue.

**Pull request merged**